src/fheroes2/ai/normal/ai_normal_hero.cpp

```diff
@@ -2095,10 +2143,28 @@ namespace AI
                 useDimensionDoor = true;
             }
 
-            if ( isTerritoryExpansion && fogDiscoveryValue < 0 ) {
-                // This is actually a very useful fog discovery action which might lead to finding of new objects.
-                // Increase the value of this action.
-                fogDiscoveryValue /= 2;
+            if ( isTerritoryExpansion ) {
+                // Over time the AI should focus more on territory expansion.
+                const uint32_t period = getTimeoutBeforeFogDiscoveryIntensification( hero );
+                assert( period > 0 );
+
+                if ( fogDiscoveryValue < 0 ) {
+                    // This is actually a very useful fog discovery action which might lead to finding of new objects.
+                    // Increase the value of this action.
+                    fogDiscoveryValue /= 2;
+
+                    if ( world.CountDay() > period ) {
+                        fogDiscoveryValue = 0;
+                    }
+                    else {
+                        fogDiscoveryValue = fogDiscoveryValue * ( period - world.CountDay() ) / period;
+                    }
```

Comment:
For the case when `world.CountDay() > period` the `/= 2` is redundant because the variable will be set to 0.
So we can slightly increase the performance:

```suggestion
                    if ( world.CountDay() > period ) {
                        fogDiscoveryValue = 0;
                    }
                    else {
                        // Increase the value of this action.
                        fogDiscoveryValue /= 2;
                        
                        fogDiscoveryValue = fogDiscoveryValue * ( period - world.CountDay() ) / period;
                    }
```

Reply:
@Districh-ru  If so, there is no need to divide it by 2 first and then perform separate calculations, we can just write like this:

```cpp
else {
    fogDiscoveryValue = fogDiscoveryValue / 2 * ( period - world.CountDay() ) / period;
}
```

Reply:
Sure, I just left the division with the comment why the division is done from the previous code.